### ISSUE TITLE
File Manager: Stopped working (Cloudflare Apps)

### ISSUE THREAD

**Issue by TheVukasLt** [qooob/authentic-theme#511]
Hey,
haven't noticed any update, so, not sure but yesterday the file manager stopped working.

Tried removing and reinstalling theme but worthless. No issue with other themes (default).

Here is the issue: http://imgur.com/Bw0yc8U

First screen loads as is (as you can see in pic) however I can not get into any folder by neither clicking on it what just changes view in red block nor opening in new tab what, again, changes red block view. Clicking inside red block just refresh the page.
So, basically it's useless now.

**Comment by qooob** [comment]
Hi,

What happens if you log in as `root`? `root_tvl` seems to be causing an issue. Have you done any recent changes on this user previously in case it worked before? Is this user in wheel group? Is this system user or user created by Webmin ACL?

**Comment by TheVukasLt** [comment]
qooob,

No, no changes to any user since server install and set up. I cannot use root as it is the primary user created with key, so it has no password, so no way to log in with it. In order to have a login I created a new user with command line and gained sudo rights, that's all. 

However I'll give a try creating a new sudo user to see if it fixes.

**Comment by qooob** [comment]
No, can you go to Webmin->Webmin Users and create a user there. Don't forget to grant access to needed modules.

**Comment by TheVukasLt** [comment]
Have never tried before so let's give it a try now. :)

**Comment by TheVukasLt** [comment]
Sadly, nothing changes.

**Comment by TheVukasLt** [comment]
Oh, the web console, haven't noticed this before:
```
Refused to load the script 'https://ajax.cloudflare.com/cdn-cgi/nexp/dok3v=e982913d31/cloudflare.min.js' because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-inline' 'unsafe-eval'".
(anonymous function) @ panel.[my_domain_here]/:8
parent.min.js?180301:6 Welcome to Authentic Theme 18.03 https://github.com/qooob/authentic-theme
:8443/sysinfo.cgi:8 Refused to load the script 'https://ajax.cloudflare.com/cdn-cgi/nexp/dok3v=e982913d31/cloudflare.min.js' because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-inline' 'unsafe-eval'".
(anonymous function) @ :8443/sysinfo.cgi:8
:8443/acl/:8 Refused to load the script 'https://ajax.cloudflare.com/cdn-cgi/nexp/dok3v=e982913d31/cloudflare.min.js' because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-inline' 'unsafe-eval'".
(anonymous function) @ :8443/acl/:8
:8443/acl/edit_user.cgi?user=thevukaslt:8 Refused to load the script 'https://ajax.cloudflare.com/cdn-cgi/nexp/dok3v=e982913d31/cloudflare.min.js' because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-inline' 'unsafe-eval'".
(anonymous function) @ :8443/acl/edit_user.cgi?user=thevukaslt:8
:8443/filemin/:8 Refused to load the script 'https://ajax.cloudflare.com/cdn-cgi/nexp/dok3v=e982913d31/cloudflare.min.js' because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-inline' 'unsafe-eval'".
(anonymous function) @ :8443/filemin/:8
https://panel.[my_domain_here]:8443/images/icons/mime/application-octet-stream.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/inode-mount-point.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/inode-directory.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/text-plain.png Failed to load resource: the server responded with a status of 404 ()
2https://panel.[my_domain_here]:8443/images/icons/mime/inode-symlink.png Failed to load resource: the server responded with a status of 404 ()
2https://panel.[my_domain_here]:8443/images/icons/mime/inode-directory.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/inode-symlink.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/inode-directory.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/inode-symlink.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/text-plain.png Failed to load resource: the server responded with a status of 404 ()
https://panel.[my_domain_here]:8443/images/icons/mime/application-octet-stream.png Failed to load resource: the server responded with a status of 404 ()
```

**Comment by qooob** [comment]
It seems that you're using some browsers extensions?  Can you try in incognito tab or with disabled extensions? 😉

**Comment by qooob** [comment]
Do you use Webmin's inbuilt webserver or something else?

**Comment by TheVukasLt** [comment]
Well, I have few extensions on latest Chrome 51, but incognito doesn't help.

Not totally sure about last question, so, to be clear, my installation process was definitely the same as provided in this article:
https://www.digitalocean.com/community/tutorials/how-to-install-webmin-with-ssl-on-ubuntu-14-04
til topic "Log in to Webmin".

Just that later I manually (by editing miniserv.conf) changed the port from 10000 to 8443 in order to comply with Cloudflare's enabled ports.

While doubting about CDN, I will go ahead and try disabling Cloudflare for this domain to see if it fixes.

On the other hand totally removing Webmin and freshly installing it again through command line is still an option and while if it would fix it (hope so), then we wouldn't know the reason for this...

**Comment by qooob** [comment]
Changing port is alright. Not sure about Cloudflare but it doesn't seem as theme issue at all.

**Comment by TheVukasLt** [comment]
UPDATE: Instead waiting for CDN to terminate services on domain I went ahead and logged with an IP address of server and Volia! It works.

**Comment by TheVukasLt** [comment]
Alright, here we go. I made few tests and confirmed it twice. 

The core of the problem was Cloudflare's App called "A better browser". Some *.js from Cloudflare are being injected into page in order to run this app and seems they cause interference.

Solution: either disable this app or create a new Page Rule that disables apps on your Webmin web server.

Here is a visual example: http://imgur.com/gallery/0JM28

Cheers! :)

**Comment by qooob** [comment]
Yes, sounds correct. Thanks for sharing.

**Comment by TheVukasLt** [comment]
You welcome, and to note that all errors from web console went away after applying solution.
